### PR TITLE
is_enterprise/is_dedicated for Python

### PR DESCRIPTION
**Pull request by otaviojava**

### CODE REVIEW

File: platformshconfig/config.py [hunk 1]
```diff
@@ -376,8 +376,8 @@ def application(self):
             )
         return self._applicationDef
 
-    def on_enterprise(self):
-        """Determines if the current environment is a Platform.sh Enterprise environment.
+    def on_dedicated(self):
+        """Determines if the current environment is at Platform.sh Enterprise environment.
```
Comment (Crell) [inline platformshconfig/config.py]:
```suggestion
        """Determines if the current environment is a Platform.sh Dedicated environment.
```

File: platformshconfig/config.py [hunk 2]
```diff
@@ -387,6 +387,18 @@ def on_enterprise(self):
...
         return self.is_valid_platform() and self['MODE'] == 'enterprise'
 
+    def on_enterprise(self):
+        """Determines if the current environment is at Platform.sh Enterprise environment.
+           Depreated, use on_dedicated instead
+
+        Returns:
+            bool:
+                True on an Enterprise environment, False otherwise.
+
```
Comment (Crell) [inline platformshconfig/config.py]:
```suggestion
```

File: platformshconfig/config.py [hunk 3]
```diff
@@ -387,6 +387,18 @@ def on_enterprise(self):
 
         return self.is_valid_platform() and self['MODE'] == 'enterprise'
 
+    def on_enterprise(self):
+        """Determines if the current environment is at Platform.sh Enterprise environment.
+           Depreated, use on_dedicated instead
+
+        Returns:
+            bool:
+                True on an Enterprise environment, False otherwise.
```
Comment (Crell) [inline platformshconfig/config.py]:
```suggestion
                True on a Dedicated environment, False otherwise.
```

**Comment by Crell** [comment]
Tests and README still need updating, plus a CHANGELOG.

**Comment by chadwcarlson** [comment]
* Bump the version to `2.3.1` in `setup.py`
* Add Changelog in slack
* Update README with note to match others

File: README.md [hunk 4]
```diff
@@ -62,9 +62,17 @@ config.in_runtime()
 
 config.on_enterprise()
```
Comment (Crell) [inline README.md]: The on_enterprise() method should be removed from the README.

File: platformshconfig/config.py [hunk 5]
```diff
@@ -362,8 +362,10 @@ def get_route(self, route_id):
     def application(self):
         """Returns the application definition dict.
 
-        This is, approximately, the .platform.app.yaml file as a nested dict. However, it also has other information
-        added by Platform.sh as part of the build and deploy process.
+           @deprecated
```
Comment (Crell) [inline platformshconfig/config.py]: To be fair, I don't know if @deprecated is as common a marker in Python as it is in PHP and JS and other JavaDoc-inpsired languages.  I'm good with using it but there may be some other Python specific convention.

**Pull request merged**